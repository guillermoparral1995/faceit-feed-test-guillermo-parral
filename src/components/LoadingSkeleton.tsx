const SkeletonPost = () => {
  return (
    <article className="animate-pulse flex gap-4 max-w-3xl mx-auto dark:prose-dark rounded-md border-2 border-slate-400 p-4 my-4">
      <div className="flex-none">
        <div className="w-10 h-10 bg-slate-400 rounded-sm"></div>
      </div>
      <div className="flex-auto">
        <div className="w-20 h-4 bg-slate-400 rounded-sm mb-2"></div>
        <div className="w-80 h-4 bg-slate-400 rounded-sm my-1"></div>
        <div className="w-40 h-4 bg-slate-400 rounded-sm my-1"></div>
      </div>
    </article>
  );
};

export default function LoadingSkeleton() {
  return (
    <>
      <SkeletonPost />
      <SkeletonPost />
      <SkeletonPost />
    </>
  );
}
