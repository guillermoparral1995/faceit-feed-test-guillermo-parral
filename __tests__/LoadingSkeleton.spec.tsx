import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import LoadingSkeleton from "@/components/LoadingSkeleton";

describe("LoadingSkeleton", () => {
  it("should render correctly", () => {
    const component = render(<LoadingSkeleton />);

    expect(component).toMatchSnapshot();
  });
});
