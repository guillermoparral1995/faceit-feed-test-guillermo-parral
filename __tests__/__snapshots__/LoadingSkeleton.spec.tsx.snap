// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSkeleton should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="animate-pulse flex gap-4 max-w-3xl mx-auto dark:prose-dark rounded-md border-2 border-slate-400 p-4 my-4"
        data-testid="skeleton"
      >
        <div
          class="flex-none"
        >
          <div
            class="w-10 h-10 bg-slate-400 rounded-sm"
          />
        </div>
        <div
          class="flex-auto"
        >
          <div
            class="w-20 h-4 bg-slate-400 rounded-sm mb-2"
          />
          <div
            class="w-80 h-4 bg-slate-400 rounded-sm my-1"
          />
          <div
            class="w-40 h-4 bg-slate-400 rounded-sm my-1"
          />
        </div>
      </article>
      <article
        class="animate-pulse flex gap-4 max-w-3xl mx-auto dark:prose-dark rounded-md border-2 border-slate-400 p-4 my-4"
        data-testid="skeleton"
      >
        <div
          class="flex-none"
        >
          <div
            class="w-10 h-10 bg-slate-400 rounded-sm"
          />
        </div>
        <div
          class="flex-auto"
        >
          <div
            class="w-20 h-4 bg-slate-400 rounded-sm mb-2"
          />
          <div
            class="w-80 h-4 bg-slate-400 rounded-sm my-1"
          />
          <div
            class="w-40 h-4 bg-slate-400 rounded-sm my-1"
          />
        </div>
      </article>
      <article
        class="animate-pulse flex gap-4 max-w-3xl mx-auto dark:prose-dark rounded-md border-2 border-slate-400 p-4 my-4"
        data-testid="skeleton"
      >
        <div
          class="flex-none"
        >
          <div
            class="w-10 h-10 bg-slate-400 rounded-sm"
          />
        </div>
        <div
          class="flex-auto"
        >
          <div
            class="w-20 h-4 bg-slate-400 rounded-sm mb-2"
          />
          <div
            class="w-80 h-4 bg-slate-400 rounded-sm my-1"
          />
          <div
            class="w-40 h-4 bg-slate-400 rounded-sm my-1"
          />
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="animate-pulse flex gap-4 max-w-3xl mx-auto dark:prose-dark rounded-md border-2 border-slate-400 p-4 my-4"
      data-testid="skeleton"
    >
      <div
        class="flex-none"
      >
        <div
          class="w-10 h-10 bg-slate-400 rounded-sm"
        />
      </div>
      <div
        class="flex-auto"
      >
        <div
          class="w-20 h-4 bg-slate-400 rounded-sm mb-2"
        />
        <div
          class="w-80 h-4 bg-slate-400 rounded-sm my-1"
        />
        <div
          class="w-40 h-4 bg-slate-400 rounded-sm my-1"
        />
      </div>
    </article>
    <article
      class="animate-pulse flex gap-4 max-w-3xl mx-auto dark:prose-dark rounded-md border-2 border-slate-400 p-4 my-4"
      data-testid="skeleton"
    >
      <div
        class="flex-none"
      >
        <div
          class="w-10 h-10 bg-slate-400 rounded-sm"
        />
      </div>
      <div
        class="flex-auto"
      >
        <div
          class="w-20 h-4 bg-slate-400 rounded-sm mb-2"
        />
        <div
          class="w-80 h-4 bg-slate-400 rounded-sm my-1"
        />
        <div
          class="w-40 h-4 bg-slate-400 rounded-sm my-1"
        />
      </div>
    </article>
    <article
      class="animate-pulse flex gap-4 max-w-3xl mx-auto dark:prose-dark rounded-md border-2 border-slate-400 p-4 my-4"
      data-testid="skeleton"
    >
      <div
        class="flex-none"
      >
        <div
          class="w-10 h-10 bg-slate-400 rounded-sm"
        />
      </div>
      <div
        class="flex-auto"
      >
        <div
          class="w-20 h-4 bg-slate-400 rounded-sm mb-2"
        />
        <div
          class="w-80 h-4 bg-slate-400 rounded-sm my-1"
        />
        <div
          class="w-40 h-4 bg-slate-400 rounded-sm my-1"
        />
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
